lisa/sut_orchestrator/baremetal/cluster/idrac.py

```diff
@@ -214,30 +240,139 @@ def _wait_for_completion(self, response: Any, timeout: int = 600) -> None:
                 task = response.monitor(self.redfish_instance)
 
         if response.status not in [200, 202, 204]:
-            raise LisaException("Failed to complete task! - status:", response.status)
-
-    def _insert_virtual_media(self, iso_http_url: str) -> None:
-        self._log.debug("Inserting virtual media...")
-        body = {"Image": iso_http_url}
+            # Include response details for better debugging
+            error_msg = f"iDRAC API request failed with status {response.status}"
+            if hasattr(response, "text"):
+                error_msg += f", response: {response.text}"
+            if hasattr(response, "dict"):
+                error_msg += f", details: {response.dict}"
+            raise LisaException(error_msg)
+
+    def _eject_vm(self, device_name: str) -> None:
+        """Eject virtual media from specified device (CD or RemovableDisk)."""
         response = self.redfish_instance.post(
-            "/redfish/v1/Managers/iDRAC.Embedded.1/VirtualMedia/CD/Actions/"
-            "VirtualMedia.InsertMedia",
-            body=body,
+            f"/redfish/v1/Managers/iDRAC.Embedded.1/VirtualMedia/{device_name}/"
+            "Actions/VirtualMedia.EjectMedia",
+            body={},
         )
-        self._wait_for_completion(response)
-        self._log.debug("Inserting virtual media completed...")
+        # Ignore errors - device may not have media inserted
+        if response.status in [200, 202, 204]:
+            self._wait_for_completion(response)
+
+    def _get_vm_state(self, device_name: str = "CD") -> Dict[str, Any]:
+        """Get virtual media state for specified device."""
+        response = self.redfish_instance.get(
+            f"/redfish/v1/Managers/iDRAC.Embedded.1/VirtualMedia/{device_name}"
+        )
+        result: Dict[str, Any] = response.dict
```

Comment:
Please return typed result, so it's easy to use and troubleshooting. So far, I saw only "Inserted" is used.